src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,223 @@
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndexGet(index);
+        var n = getNodeByIndex(index).item;
+        remove(index);
+        add(value,index);
+        return n;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndexGet(index);
+        Node<T> removeResult = getNodeByIndex(index);
+        if (index == 0) {
+            return unlinkFirst(removeResult);
+        }
+        if (index == size) {
+            return unlinkLast(removeResult);
+        }
+        return unlink(removeResult);
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (first == null);
+    }
+
+    private static class Node<T> {
+        private T item;
+        private MyLinkedList.Node<T> next;
+        private MyLinkedList.Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, MyLinkedList.Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkPositionIndex(index);
+        if (index < (size >> 1)) {
+            var x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            var x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!isPositionIndex(index)) {
+            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private void checkPositionIndexGet(int index) {
+        if (!isPositionIndexGet(index)) {
+            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private boolean isPositionIndex(int index) {
+        return index >= 0 && index <= size;
+    }
+
+    private boolean isPositionIndexGet(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private String outOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+    private T unlinkFirst(MyLinkedList.Node<T> first) {
```

Comment:
it's enough to have one unlink method, you have three, rewrite your logic, so you have only one